dot/sync/chain_sync.go

```diff
@@ -783,6 +785,10 @@ taskResultLoop:
 }
 
 func (cs *chainSync) handleReadyBlock(bd *types.BlockData, origin blockOrigin) error {
+	if bd == nil {
+		logger.Debugf("nil block data provided to handleReadyBlock")
+		return fmt.Errorf("nil block data provided to handleReadyBlock")
+	}
```

Comment:
instead of handling this bd nil here, can we detect it at `handleWorkersResults` at the loop  where we insert the block data in the `syncingChain` slice, and if the `blockInResponse` is `nil` we create and dispatch a request to acquire it?

Reply:
Maybe you can add in another PR

Reply:
good call, modifying to add in this PR. Let me know how it looks